core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -199,6 +199,7 @@ public synchronized void connect() throws AlluxioStatusException {
     disconnect();
     Preconditions.checkState(!mClosed, "Client is closed, will not try to connect.");
 
+    mContext.getUserState().relogin();
```

Comment:
Can you put this to `beforeConnect()` ?

Reply:
beforeConnect() would be too late. The relogin() must happen before getAddress() which is called in the try block before the beforeConnect().